jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1870,7 +1875,10 @@ else if (scheme.equalsIgnoreCase("file"))
      * @param resource If the string starts with one of the ALLOWED_SCHEMES, then it is assumed to be a
      * representation of a {@link URI}, otherwise it is treated as a {@link Path}.
      * @return The {@link URI} form of the resource.
+     * @deprecated This method is currently resolving relative paths against the current directory, which is a mechanism
+     * that should be implemented by a {@link ResourceFactory}.   All calls to this method need to be reviewed.
      */
+    @Deprecated(since = "12.0.8")
```

Comment:
Do we really want to deprecate this method? It still serves the purpose of converting a string into a URI, and is still useful when the string contains an absolute path.

Reply:
A String can be converted into a URI with `new Uri(string)`, so it is not necessary

Reply:
No, `new URI(String)` is not safe to convert a String to a URI.
There are many examples of that, esp on Windows.

```
jshell> var u2 = new URI("C:\\code\\base")
|  Exception java.net.URISyntaxException: Illegal character in opaque part at index 2: C:\code\base
|        at URI$Parser.fail (URI.java:2974)
|        at URI$Parser.checkChars (URI.java:3145)
|        at URI$Parser.parse (URI.java:3181)
|        at URI.<init> (URI.java:623)
|        at do_it$Aux (#2:1)
|        at (#2:1)
```

```
jshell> var u1 = URI.create("C:\\code\\base")
|  Exception java.lang.IllegalArgumentException: Illegal character in opaque part at index 2: C:\code\base
|        at URI.create (URI.java:906)
|        at do_it$Aux (#1:1)
|        at (#1:1)
|  Caused by: java.net.URISyntaxException: Illegal character in opaque part at index 2: C:\code\base
|        at URI$Parser.fail (URI.java:2974)
|        at URI$Parser.checkChars (URI.java:3145)
|        at URI$Parser.parse (URI.java:3181)
|        at URI.<init> (URI.java:623)
|        at URI.create (URI.java:904)
|        ...
```

Reply:
@joakime Those cases are precisely what `ResourceFactory` is for.  It is the class that works out what the string is and somehow makes a resource for it.   
The `URIUtil.toURI` method was handling the non-URI by resolving against the current directory in the file system, but who knows what happens if it was a jar:file.

If you don't know what the string is, then use a `ResourceFactory`.  If you know the string is a URI, then use `new URI(String)`

In #11577, I have been able to replace all the `toURI` usages with `ResourceFactory`.